### PR TITLE
feat: rework ediSplitter into metadata extractor

### PR DESCRIPTION
**Pull request by BDQ**
This PR repurposes the original ediSplitter to now only provide metadata about a files Interchanges.

The [inbound/handler](https://github.com/Stedi-Demos/bootstrap/compare/feat/edi-metadata?expand=1#diff-8358646551c8d5bf18791ef3b770c0f60b55e1c8465760cc41d6702f8e6bbf5aR84) then iterates over the metadata making single ST only calls to Translate.

The metadata extractor no longer enforces Translates rules (like only a single transaction set type per group), and doesn't create an artificial interchanges to send to Translate.

It will not produce a object like:

```js
{
        senderId: "12/7147085121",
        receiverId: "01/040132628",
        controlNumber: 501,
        delimiters: {
          segment: "~",
          element: "*",
        },
        segments: {
          ISA: "ISA*00*          *00*          *12*7147085121     *01*040132628      *220921*1002*U*00501*000028538*0*P*>",
          IEA: "IEA*1*000028538",
        },
        functionalGroups: [
          {
            controlNumber: 1002,
            release: "005010",
            applicationSenderCode: "7147085121",
            applicationReceiverCode: "040132628",
            segments: {
              GS: "GS*PR*7147085121*040132628*20220921*1002*28538*X*005010",
              GE: "GE*8*28538",
            },
            transactionSets: [
              {
                segments: [
                  "ST*855*0001",
                  "BAK*00*AK*4179213*20220920",
                  "SE*11*0001",
                ],
                controlNumber: 1,
                transactionSetType: "855",
              },
            ],
          },
          {
            controlNumber: 946,
            release: "005010",
            applicationSenderCode: "009599671",
            applicationReceiverCode: "040132628",
            segments: {
              GS: "GS*FA*009599671*040132628*20220926*0946*7540*X*005010",
              GE: "GE*8*28538",
            },
            transactionSets: [
              {
                segments: [
                  "ST*997*11445",
                  "AK1*AO*00000000",
                  "AK9*X*000000*00*00",
                  "SE*11*0001",
                ],

                controlNumber: 11445,
                transactionSetType: "997",
              },
              {
                segments: [
                  "ST*997*11446",
                  "AK1*AO*00000000",
                  "AK9*X*000000*00*00",
                  "SE*11*0001",
                ],
                controlNumber: 11446,
                transactionSetType: "997",
              },
              {
                segments: [
                  "ST*999*0246",
                  "AK1*AO*00000000",
                  "AK9*X*000000*00*00",
                  "SE*11*0001",
                ],
                controlNumber: 246,
                transactionSetType: "999",
              },
            ],
          },
        ],
      }
```

### CODE REVIEW

File: src/lib/prepareMetadata.ts [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+
+        currentTransactionSet = {
+          transactionSetType,
+          controlNumber: parseInt(elements[2]),
+          segments: [currentSegment],
+        };
+
+        break;
+      case "SE":
+        if (!currentFunctionalGroup)
```
Comment (rsioss) [inline src/lib/prepareMetadata.ts:110]: oh, related to the comment above, looks like maybe this just needs to be moved up

**Review by rsioss — APPROVED** [review]
This is a nice improvement over the previous implementation! 🚀 

Left a number of nit comments... nothing blocking, and can certainly be done in a follow-up if you prefer. I also noticed there are several formatting changes, maybe we should set up `prettier` for the repo so we're all using the same style?

**Comment by BDQ** [comment]
> This is a nice improvement over the previous implementation! 🚀
> 
> I also noticed there are several formatting changes, maybe we should set up `prettier` for the repo so we're all using the same style?

Yes, good call on prettier

File: src/functions/edi/inbound/handler.ts [hunk 2]
```diff
@@ -79,72 +79,88 @@ export const handler = async (event: any): Promise<Record<string, any>> => {
...
+            const receivingPartnerId = await resolvePartnerIdFromISAId(
+              interchange.receiverId
             );
 
-            const ediProcessingResult = await processEdiDocument(
-              guideSummary.guideId,
-              ediDocument.edi,
-              mappingId
+            // load the Partnership for the sending and receiving partners
+            const partnership = await loadPartnership(
```
[the remark this answers is not in the document]
Comment (BDQ) [reply src/functions/edi/inbound/handler.ts:95]: you only need to await on a for loop when the iterable after the `of` is async.

File: src/lib/prepareMetadata.ts [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+        currentFunctionalGroup = {
+          release,
+          controlNumber: parseInt(elements[5]),
+          applicationSenderCode: elements[2],
+          applicationReceiverCode: elements[3],
+          transactionSets: [],
+          segments: { GS: currentSegment },
+        };
+        break;
+      case "ST":
```
[the remark this answers is not in the document]
Comment (BDQ) [reply src/lib/prepareMetadata.ts:96]: It's not needed for types, but I'll add it again for completeness.

File: src/functions/edi/inbound/handler.ts [hunk 4]
```diff
@@ -79,72 +79,88 @@ export const handler = async (event: any): Promise<Record<string, any>> => {
...
+
+            // )
+
+            // get transaction set configs for partnership
+            const transactionSetConfigs =
+              getTransactionSetConfigsForPartnership({
+                partnership,
+                sendingPartnerId,
+                receivingPartnerId,
+              });
```
[the remark this answers is not in the document]
Comment (BDQ) [reply src/functions/edi/inbound/handler.ts:111]: I'm not going to flatten the functional groups loop - I think it'll help the reader appreciate the structure a little easier.

File: src/functions/edi/inbound/handler.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (rsioss) [reply src/functions/edi/inbound/handler.ts]: > you only need to await on a for loop when the iterable after the of is async.

ahh, the iterable itself, doesn't matter whether or not there's an async call in the loop, makes sense! 💡

**Review by rsioss — APPROVED** [review]
> @rsioss I address your feedback, do you mind giving it another test (works for me)?

@BDQ looks great and works for me as well! 🚀 🚢

**Pull request merged**